pints/_mcmc/_slice_hyperrectangles.py

```diff
@@ -0,0 +1,280 @@
+            # proposed sample for next iteration
+            self._current = np.array(self._x0, copy=True)
+            self._current_log_pdf = fx
+            self._proposed = np.array(self._current, copy=True)
+
+            # Sample height of the slice log_y for next iteration
+            e = np.random.exponential(1)
+            self._current_log_y = self._current_log_pdf - e
+
+            # Return first point in chain, which is x0
+            return np.array(self._current, copy=True)
+
+        # Subsequent calls
+        if self._current_log_y < fx:
+            # The accepted sample becomes the new current sample
+            self._current = np.array(self._proposed, copy=True)
+
+            # Sample new log_y used to define the next slice
+            e = np.random.exponential(1)
+            self._current_log_y = fx - e
+
+            self._hyperrectangle_positioned = False
+
+            # Return accepted sample
+            return np.array(self._proposed, copy=True)
+
+        # Shrinking
+        else:
+            # Adaptive shrinking: shrink in the direction ``index``
+            # in which ``(R_i - L_i) |G_i|`` is maximised
+            if self._adaptive:
+                # Store products ``(R_i - L_i) |G_i|``
+                temp = np.zeros(self._n_parameters)
+                for i in range(self._n_parameters):
+                    temp[i] = (self._R[i] - self._L[i]) * np.abs(grad[i])
+
+                # Index which maximises ``(R_i - L_i) |G_i|``
+                index = np.argmax(temp)
+
+                # Shrink only in the direction ``index``
+                if self._proposed[index] < self._current[index]:
+                    self._L[index] = self._proposed[index]
+                else:
+                    self._R[index] = self._proposed[index]
+
+            # Shrink homogeneously in all directions
+            else:
+                for i, x_1 in enumerate(self._proposed):
+                    if x_1 < self._current[i]:
+                        self._L[i] = x_1
+                    else:
+                        self._R[i] = x_1
+
+    def needs_sensitivities(self):
+        """ See :meth:`pints.MCMCSampler.needs_sensitivities()`. """
+        return True
+
+    def name(self):
+        """ See :meth:`pints.MCMCSampler.name()`. """
+        return 'Slice Sampling - Hyperrectangles'
+
+    def set_w(self, w):
+        """
+        Sets scale vector ``w`` for generating the hyperrectangle.
+        """
+        if type(w) == int or float:
+            w = np.full((len(self._x0)), w)
+        else:
+            w = np.asarray(w)
+        if any(n < 0 for n in w):
+            raise ValueError("""Width ``w`` must be positive for
+                            interval expansion.""")
+        self._w = w
+
+    def set_adaptive_shrinking(self, adaptive):
+        """
+        Sets adaptive method for shrinking the hyperrectangle.
+        """
+        if type(adaptive) != bool:
```

Comment:
This check isn't necessary! `adaptive` can be any type the user likes, as long as we can evaluate it with `if adaptive: ...`. If you really want you can just add a line like `self._adaptive = bool(adaptive)`